app/models/workflow_step.rb

```diff
@@ -26,7 +26,7 @@ class WorkflowStep < ApplicationRecord
   # @return [Nokogiri::XML::Builder::NodeBuilder]
   def as_milestone(xml)
     xml.milestone(lifecycle,
-                  date: updated_at.to_time.iso8601,
+                  date: created_at.to_time.iso8601,
```

Comment:
I believe the created_at timestamps get set when the workflow is initialized for an object.  This not the same as when the step is completed, which is what the updated_at gets.

Reply:
How about we create a new column for "completed_at" that is only set when a process is completed.  The default value can be either updated_at or created_at.

Reply:
That would make sense.  And then the milestones API could return this "completed_at" date. This gets us away from the updated_at column, which can be inadvertently touched by rails when any other column is touched.

Reply:
Sounds good to me. There are some steps where the completed_at will be noticeably later than the created_at when the object has a lot of data.